### PR TITLE
provider/digitalocean: Fixing panic condition on FloatingIP droplet

### LINKED ISSUE #4162: Panic if Digital Ocean floating IP not assigned to droplet
`terraform plan` panics if a floating IP exists, but is not currently assigned to a droplet. 
I've posted a crash log here: https://gist.github.com/franklinhu/5b116f159d491b7550ee

Thanks!

### PR DESCRIPTION
**Pull request by stack72**
Fixes #4162
Fixing the digitalocean floatingip resource for a panic when droplet wasn't available

### CODE REVIEW

**Comment by jen20** [comment]
LGTM. Thanks @stack72!

File: builtin/providers/digitalocean/resource_digitalocean_floating_ip.go [hunk 1]
```diff
@@ -83,8 +83,10 @@ func resourceDigitalOceanFloatingIpRead(d *schema.ResourceData, meta interface{}
 	}
 
 	if _, ok := d.GetOk("droplet_id"); ok {
-		log.Printf("[INFO] The region of the Droplet is %s", floatingIp.Droplet.Region)
-		d.Set("region", floatingIp.Droplet.Region.Slug)
+		if floatingIp.Droplet != nil {
+			log.Printf("[INFO] The region of the Droplet is %s", floatingIp.Droplet.Region)
+			d.Set("region", floatingIp.Droplet.Region.Slug)
+		}
```
Comment (franklinhu) [inline builtin/providers/digitalocean/resource_digitalocean_floating_ip.go]: Out of curiosity, this skips setting the region at all if the floating IP exits but the droplet doesn't. It seems like we could fall back to the below else case, but not sure what the ramifications are.

Comment (jen20) [inline builtin/providers/digitalocean/resource_digitalocean_floating_ip.go]: :+1: good point, we should probably check this explicitly.

Comment (stack72) [inline builtin/providers/digitalocean/resource_digitalocean_floating_ip.go]: @franklinhu the current tests for this are for both region based floatingips and floatingips with droplets. A floatingIP **must** either be created for a region OR a droplet. So if we fall back to the region then that seems to be the correct course of action

I haven't been able to trigger the panic

Comment (stack72) [inline builtin/providers/digitalocean/resource_digitalocean_floating_ip.go]: @franklinhu ok, I just sent #4216. To be more explicit. I don't rely on the d.GetOk anymore. I use the API result directly and then check for a droplet and use the region that way.

Comment (franklinhu) [inline builtin/providers/digitalocean/resource_digitalocean_floating_ip.go]: Ah awesome thanks!

**Pull request merged**